packages/venia-concept/src/components/ProductImageCarousel/carousel.js

```diff
@@ -1,136 +1,124 @@
     left: ChevronLeftIcon,
     right: ChevronRightIcon
 };
 
-class Carousel extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string,
-            currentImage: PropTypes.string,
-            imageContainer: PropTypes.string,
-            'chevron-left': PropTypes.string,
-            'chevron-right': PropTypes.string
-        }),
-        images: PropTypes.arrayOf(
-            PropTypes.shape({
-                label: PropTypes.string,
-                position: PropTypes.number,
-                disabled: PropTypes.bool,
-                file: PropTypes.string.isRequired
-            })
-        ).isRequired
-    };
-
-    state = {
-        activeItemIndex: 0,
-        currentImageLoaded: false
-    };
-
-    updateActiveItemIndex = index => {
-        this.setState({ activeItemIndex: index });
-    };
-
-    // The spec does not guarantee a position parameter,
-    // so the rule will be to order items without position last.
-    // See https://github.com/magento/graphql-ce/issues/113.
-    // Memoize this expensive operation based on reference equality
-    // of the `images` array. Apollo cache should return a new array
-    // only when it does a new fetch.
-    sortAndFilterImages = memoize(items =>
-        items
-            .filter(i => !i.disabled)
-            .sort((a, b) => {
-                const aPos = isNaN(a.position) ? 9999 : a.position;
-                const bPos = isNaN(b.position) ? 9999 : b.position;
-                return aPos - bPos;
-            })
-    );
+const DEFAULT_IMAGE_WIDTH = 640;
 
-    get sortedImages() {
-        const { images } = this.props;
-        return this.sortAndFilterImages(images);
-    }
+const sortAndFilterImages = images =>
+    images.filter(i => !i.disabled).sort((a, b) => a.position - b.position);
 
-    leftChevronHandler = () => {
-        const sortedImages = this.sortedImages;
-        const { activeItemIndex } = this.state;
-        activeItemIndex > 0
-            ? this.updateActiveItemIndex(activeItemIndex - 1)
-            : this.updateActiveItemIndex(sortedImages.length - 1);
-    };
-
-    rightChevronHandler = () => {
-        const sortedImages = this.sortedImages;
-        const { activeItemIndex } = this.state;
-        this.updateActiveItemIndex((activeItemIndex + 1) % sortedImages.length);
-    };
-
-    getChevron = direction => (
-        <button
-            onClick={this[`${direction}ChevronHandler`]}
-            className={this.props.classes[`chevron-${direction}`]}
-        >
+const Chevron = ({ className, direction, onClick }) => {
+    return (
+        <button onClick={onClick} className={className}>
             <Icon src={ChevronIcons[direction]} size={40} />
         </button>
     );
+};
+
+const Carousel = props => {
+    const { images = [] } = props;
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const [activeItemIndex, setActiveItemIndex] = useState(0);
```

Comment:
The biggest change to Adrian's work is that I removed the new `useCarousel` and `useSortedImages` hooks in favor of using local state and a local utility function. We should discuss the reasoning behind whether or not this logic/state belongs in a hook or not, and if we end up deciding to revert, I can do so.

cc @jimbo @tjwiebell